src/select/select.js

```diff
@@ -98,139 +132,101 @@ class Select extends React.Component<PropsT, SelectStateT> {
-      this.closeMenu();
-    }
-  };
-
   focus() {
     if (!this.input) return;
     this.input.focus();
   }
 
-  handleTouchMove = () => {
-    // Set a flag that the view is being dragged
-    this.dragging = true;
-  };
-
-  handleTouchStart = () => {
-    // Set a flag that the view is not being dragged
-    this.dragging = false;
+  // Handle touch outside on mobile to dismiss menu, ensures that the
+  // touch target is not within the wrapper DOM node.
+  handleTouchOutside = (event: TouchEvent) => {
+    if (this.wrapper && !containsNode(this.wrapper, event.target)) {
+      this.closeMenu();
+    }
   };
 
-  handleTouchEnd = (event: Event) => {
-    // Check if the view is being dragged. In this case
-    // we don't want to fire the click event (because the user only wants to scroll)
+  // Track dragging state to filter false-positive actions where a user
+  // intends to drag/scroll the page.
+  handleTouchMove = () => (this.dragging = true);
+  handleTouchStart = () => (this.dragging = false);
+  handleTouchEnd = (event: TouchEvent) => {
     if (this.dragging) return;
-    // Fire the mouse events
     this.handleClick(event);
   };
-
-  handleTouchEndClearValue = (event: Event) => {
-    // Check if the view is being dragged. In this case
-    // we don't want to fire the click event (because the user only wants to scroll)
+  handleTouchEndClearValue = (event: TouchEvent) => {
     if (this.dragging) return;
-    // Clear the value
     this.clearValue(event);
   };
 
-  handleClickOutside = (event: Event) => {
-    if (this.isValueJustSelected) {
-      this.isValueJustSelected = false;
+  handleClick = (event: MouseEvent | TouchEvent) => {
+    if (this.props.disabled || !isClick(event) || !isLeftClick(event)) {
       return;
     }
-    const isFocused = this.state.isFocused || this.state.isPseudoFocused;
-    // $FlowFixMe
-    if (isFocused && this.wrapper && !this.wrapper.contains(event.target)) {
-      this.handleInputBlur(event);
-    }
-  };
 
-  handleClick = (event: Event) => {
-    // If the event was triggered by a mouse click and not the primary
-    // button, or if the component is disabled, ignore it
-    if (
-      this.props.disabled ||
-      // $FlowFixMe
-      (event.type === 'click' && event.button !== 0)
-    ) {
-      return;
-    }
-    // $FlowFixMe
-    if (event.target.tagName === 'INPUT') {
+    // Case comes up when text has been typed into the input field. If no text provided,
+    // the 'input' element will have essentially 0 width therefore will not be clickable.
+    // When click outside does not reset input, text provided will stay rendered after clicks away
+    // from the select component. Upon subsequent clicks on the provided text, open the dropdown
+    // menu, in addition to text edit operations.
+    if (event.target === this.input) {
+      // CHASE: not sure why this condition is here. I cannot replicate a situation where clicks
```

Comment:
I think that was to handle double click on input and toggle the dropdown. @tajo is that correct?

Reply:
Just checked the double click on this version and it seems to be broken.